tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -33,13 +34,22 @@ public TraceContext(IDatadogTracer tracer, TraceTagCollection tags = null)
         {
             var settings = tracer?.Settings;
 
+            // TODO: Environment, ServiceVersion, GitCommitSha, and GitRepositoryUrl are stored on the TraceContext
+            // even though they likely won't change for the lifetime of the process. We should consider moving them
+            // elsewhere to reduce the memory usage.
```

Comment:
Love the way this PR increases memory usage by adding new properties and passes the buck to other properties 🤣 

On a more serious note - do we _need_/want to support users changing the sha for every trace? If not, it fees like we could avoid adding these to the `TraceContext`, and instead push them _way_ down into the `AgentWriter`/`SpanBuffer` which create the `TraceChunkModel` 🤔 That would reduce the per-span allocations and I think would be very safe, as long as we _discard_ the sha/url values if a customer tries to set them at the tag level?

Reply:
Oh, you noticed? 😉 

Fixed in ad706023cb5f9f53a6a07b9fbcf16dac6ef70706. 

> as long as we discard the sha/url values if a customer tries to set them at the tag level
As per the other comments, we've removed the ability for customers to set the sha/url programmatically, at least for the time being.

Reply:
@andrewlock Note the commit I referenced earlier missed an important detail, please see 68e09526e66399e23b046ca05d57e2c9fa13a8e1